### PR TITLE
[ASM] Fix Standalone Billing tests on propagated tags

### PR DESCRIPTION
**Pull request by e-n-0**
## Motivation

The propagated tags on the ASM Standalone billing tests are not tested on the right span.
According to [this RFC](https://docs.google.com/document/d/1uLBJGxjB4zbgor4DS_8BMzRRv4gSbuQtht0z7wBNl04/edit):
> For older endpoints (v01, v02, v03, v04, v05), we set trace tags in [Meta](https://github.com/DataDog/datadog-agent/blob/541932c9555daf8ab264fb85906a6a2528b2a60e/pkg/trace/pb/span.pb.go#L53) of the first span of each trace chunk.

The dotnet tracer implement [it this way](https://github.com/DataDog/dd-trace-dotnet/pull/4332), so all propagated tags like `_dd.p.appsec` is set on the the first span of each trace chunk.

Because this implementation is not followed by every tracers, the tests need to support the both cases:
- in the first span of trace chunk
- in the local root span

## Changes

Made a helper method to test the presence of specific tag values on the first span of a trace chunk or in the local root span.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
would it make sense to make the tests fail in case of non-compliance with the rfc?for example if the tag appears in the second span of a trace
There are conflicts with the main branch
Added reviewers from this PR: https://github.com/DataDog/system-tests/pull/2630

**Review by gnufede — APPROVED** [review]

**Review by daniel-romano-DD — APPROVED** [review]
Great work

**Review by iunanua — CHANGES_REQUESTED** [review]
These tests are making nodejs fail. 
The reason could be that `interfaces.library.get_spans(request=self.r):` is returning `GET /returnheaders` request instead `GET /requestdownstream` (both have the same `"http.useragent": "system_tests rid/${ID}"`)

And that's why we introduced the `if not REQUESTDOWNSTREAM_RESOURCE_PATTERN.search(span["resource"]):` check

**Comment by iunanua** [comment]
I'm changing a bit nodejs tests to address the issue. Please wait until node PR is merged.

Thkx

File: tests/appsec/test_asm_standalone.py
Comment (jandro996) [inline tests/appsec/test_asm_standalone.py:47]: The sampling priority is also been checked in the first span

```suggestion
                assert first_trace["trace_id"] == 1212121212121212121
                assert first_trace["metrics"]["_sampling_priority_v1"] < 2
```

File: tests/appsec/test_asm_standalone.py
Comment (jandro996) [inline tests/appsec/test_asm_standalone.py:53]:
```suggestion

```

**Comment by jandro996** [comment]
This changes also breaks some tests in java, is more or less related with what I've reported in the java PR related with the sampling priority metric. 
The main problem is that we are facing issues not related with this ASM Standalone billing RFC scope, for instance in java the propagation tags (all of them not only our new one _dd.p_appsec) doesn't follow the  **all propagated tags like are set on the the first span of each trace chunk** rule

**Comment by jandro996** [comment]
All the `assert trace[0]["metrics"]["_sampling_priority_v1"]` can be moved to the first_trace asserts 😃

File: tests/appsec/test_asm_standalone.py
Comment (e-n-0) [reply tests/appsec/test_asm_standalone.py:53]: At least in dotnet the sampling priority is set on the local root span and not on the first span, so this change will make the tests fail

File: tests/appsec/test_asm_standalone.py
Comment (e-n-0) [reply tests/appsec/test_asm_standalone.py:47]: Same comment as below

File: tests/appsec/test_asm_standalone.py
Comment (e-n-0) [reply tests/appsec/test_asm_standalone.py]: I updated the tests so the tag is searched on the local root span and the first span in trace chunk

**Review by robertomonteromiguel — APPROVED** [review]

**Review by jandro996 — APPROVED** [review]
LGTM, Good Job!

**Pull request merged**
